src/algebra/lattice_ordered_group.lean

```diff
@@ -0,0 +1,585 @@
+begin
+  rw neg_eq_pos_inv,
+  simp,
+end
+
+-- We use this in Bourbaki A.VI.12  Prop 9 a)
+/--
+Let `α` be a lattice ordered commutative group. For all elements `a`, `b` and `c` in `α`,
+$$c + (a ⊓ b) = (c + a) ⊓ (c + b).$$
+-/
+@[to_additive]
+lemma mul_inf_eq_mul_inf_mul (a b c : α) : c * (a ⊓ b) = (c * a) ⊓ (c * b) :=
+begin
+  rw le_antisymm_iff,
+  split,
+  { simp, },
+  { rw [← mul_le_mul_iff_left c⁻¹, ← mul_assoc, inv_mul_self, one_mul, le_inf_iff], simp, }
+end
+
+-- We use this in Bourbaki A.VI.12  Prop 9 a)
+/--
+Let `α` be a lattice ordered commutative group with identity `0` and let `a` be an element in `α`
+with negative component `a⁻`. Then
+$$a⁻ = -(a ⊓ 0).$$
+-/
+@[to_additive]
+lemma neg_eq_inv_inf_one (a : α) : a⁻ = (a ⊓ 1)⁻¹ :=
+begin
+  unfold lattice_ordered_comm_group.neg,
+  rw ← inv_inj,
+  rw inv_sup_eq_inv_inf_inv,
+  rw [inv_inv, inv_inv, one_inv],
+end
+
+-- Bourbaki A.VI.12  Prop 9 a)
+/--
+Let `α` be a lattice ordered commutative group and let `a` be an element in `α` with positive
+component `a⁺` and negative component `a⁻`. Then `a` can be decomposed as the difference of `a⁺` and
+`a⁻`.
+-/
+@[to_additive]
+lemma pos_inv_neg (a : α) : a = a⁺ / a⁻ :=
+begin
+  rw div_eq_mul_inv,
+  apply eq_mul_inv_of_mul_eq,
+  unfold lattice_ordered_comm_group.neg,
+  rw [mul_sup_eq_mul_sup_mul, mul_one, mul_right_inv, sup_comm],
+  unfold lattice_ordered_comm_group.pos,
+end
+
+-- Hack to work around rewrite not working if lhs is a variable
+@[to_additive, nolint doc_blame_thm]
+lemma pos_div_neg' (a : α) :  a⁺ / a⁻ = a := (pos_inv_neg _).symm
+
+-- Bourbaki A.VI.12  Prop 9 a)
+/--
+Let `α` be a lattice ordered commutative group and let `a` be an element in `α` with positive
+component `a⁺` and negative component `a⁻`. Then `a⁺` and `a⁻` are co-prime (and, since they are
+positive, disjoint).
+-/
+@[to_additive]
+lemma pos_inf_neg_eq_one (a : α) : a⁺ ⊓ a⁻=1 :=
+begin
+  rw [←mul_right_inj (a⁻)⁻¹, mul_inf_eq_mul_inf_mul, mul_one, mul_left_inv, mul_comm,
+    ←div_eq_mul_inv, pos_div_neg', neg_eq_inv_inf_one, inv_inv],
+end
+
+-- Bourbaki A.VI.12 (with a and b swapped)
+/--
+Let `α` be a lattice ordered commutative group, let `a` and `b` be elements in `α`, and let
+`(a - b)⁺` be the positive componet of `a - b`. Then
+$$a⊔b = b + (a - b)⁺.$$
+-/
+@[to_additive]
+lemma sup_eq_mul_pos_div (a b : α) : a ⊔ b = b * (a / b)⁺ :=
+  calc  a ⊔ b = (b * (a / b)) ⊔ (b * 1) : by {rw [mul_one b, div_eq_mul_inv, mul_comm a,
+    mul_inv_cancel_left], }
+  ... = b * ((a / b) ⊔ 1) : by { rw ← mul_sup_eq_mul_sup_mul (a / b) 1 b}
+
+-- Bourbaki A.VI.12 (with a and b swapped)
+/--
+Let `α` be a lattice ordered commutative group, let `a` and `b` be elements in `α`, and let
+`(a - b)⁺` be the positive componet of `a - b`. Then
+$$a⊓b = a - (a - b)⁺.$$
+-/
+@[to_additive]
+lemma inf_eq_div_pos_div (a b : α) : a ⊓ b = a / (a / b)⁺ :=
+  calc a ⊓ b = (a * 1) ⊓ (a * (b / a)) : by { rw [mul_one a, div_eq_mul_inv, mul_comm b,
```

Comment:
This entire calc block should be unindented. The *type* of a declaration is indented by two spaces, but proof *terms* start flush left without indentation.